### DISCUSSION TITLE
Perfomance Table using collection

### DISCUSSION THREAD

**Discussion by pirex360** [Power-Components/livewire-powergrid#430] [Q&A]
Hi, 

Thanks for your great package, I used to work with datatables.net and now I'm trying to change to your livewire component to synergise with my livewire app.

I'm using datasource collection, like this example :

```
    public function datasource(): ?Collection
    {
        $data = collect(Order::query()
                        ->select('stamp','st_tabct','st_tabof','st_tabofop', 'st_tabpr', 'data_inicio', 'nomeutil')
                        ->withNoLock()->WithUserStopData()
                        ->with('workcenter:stamp,codct,descricao')
                        ->with('operation:st_tabof,numof,prioridade,numop,descricao,cod_art,desc_art')
                        ->with('stop')
                        ->OpenStop()
                        ->get()
                    );

        return $data;
    }
```

I have only 5 records in the database, but all are slow, searching, hiding columns, sorting, etc.... 
I'm using laravel 8.x and your most recent package version and livewire.

https://user-images.githubusercontent.com/17271768/168272863-7e6e59ea-b4b7-440c-9013-f2c837fa68fb.mp4

https://user-images.githubusercontent.com/17271768/168272874-8878c697-3d31-41d0-a27f-cf071251091e.mp4

**Comment by CrazyBoy49z** [comment]
The collection is very heavy for tables, I did not extract 150k, I replaced it with a "builder" while 500k + data work fine

**Comment by pirex360** [comment]
Ok, I will try to replace the collection with a builder... and see the results.... in my previous tests, I am almost sure, the collection was faster... In the config file cache is enabled.

Regardless, in the sample I provided... there are only 5 records in the collection.

**Comment by pirex360** [comment]
Almost the same results with the Builder instead of Collection....  :(

**Comment by CrazyBoy49z** [comment]
I think the problem is in the optimization of the project, not a component

**Comment by pirex360** [comment]
Ok, I get your point...so I used another component named `rappasoft/laravel-livewire-tables`, changed the livewire component, using the same Builder... and now is blazing fast. (just a reminder that my Builder only returns 6 rows....imagine if it returns 5k)

It's a pity because I think this is a better component with more options!

I want to know if there's something I'm doing wrong to be slower... I debugged the ajax call to column order, and the time taken is small: database between 200ms and 290ms, but the UI is slow.


https://user-images.githubusercontent.com/17271768/168446933-99120db1-dcd4-446b-b756-441dafe6b57e.mp4

**Comment by luanfreitasdev** [comment]
I've seen very complex queries using Join and I haven't had any feedback about slowness.
